ocaml/xapi/repository.ml

```diff
@@ -654,6 +654,10 @@ let set_restart_device_models ~__context ~host ~kind =
       None
 
 let set_pending_guidances ~__context ~host ~guidances =
+  let pending_guidances = Db.Host.get_pending_guidances ~__context ~self:host in
```

Comment:
Would be better to add first and then remove, otherwise what if XAPI crashes in the middle here? We will have no guidances left, not old and not new (although in general XAPI's DB is not atomic to begin with...)

Reply:
The `diff` operation could get `to_be_removed` and `to_be_added`. So only the minimum changes would be applied on the DB.

Reply:
Not sure if `Db.Host.set_pending_guidances` directly would work?

Reply:
Yes, I think it works, fixed with [d197e5d](https://github.com/xapi-project/xen-api/pull/5097/commits/d197e5d17764235997fc70e4aa619076039edc3b), there is a similar usage here: 
https://github.com/xapi-project/xen-api/blob/master/ocaml/xapi/dbsync_slave.ml#L95-L96
`virtual_hardware_platform_versions` is also of type Set: https://github.com/xapi-project/xen-api/blob/master/ocaml/idl/datamodel_host.ml#L2085